### PR TITLE
Chat support reconfigurations

### PR DESCRIPTION
**Pull request by IlievaDayana**

### CODE REVIEW

File: src/hooks/foot.ts [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      ...staticProps,
+      ctaText: translations.ctaTextContent,
+      ctaHref: '/',
+      isPaymentFormVisible,
+    };
+  }
+
+  const message = lastMsg && lastMsg?.content?.find((m) => m.special);
+  if (isLastAssistantMsg && message && message?.special) {
+    return {
```
Comment (venzovisa) [inline src/hooks/foot.ts:74]: @IlievaDayana If `message` object on line 72 has value for `special` property you could skip this check on the next line `message?.special`

File: src/store/slices/chat.ts [hunk 2]
```diff
@@ -14,6 +14,7 @@ import {
 import { ChatState } from '../../interfaces/store';
 import { getQueryParam, uuidV4 } from '../../utils';
 import { chat as initialState } from '../initialState';
+// import { getExtractedData } from '../../utils/formatting';
```
Comment (venzovisa) [inline src/store/slices/chat.ts:18]: @IlievaDayana Do you really need this import?

**Review by venzovisa — APPROVED** [review]

File: src/hooks/foot.ts (hunk 1 above)
Comment (IlievaDayana) [reply src/hooks/foot.ts:74]: No, typescript complains, maybe its all because the prop is optional

File: __tests__/feature/chat-history.spec.jsx [hunk 3]
```diff
@@ -10,16 +10,65 @@ import renderWithProviders from '../../src/utils/storeMockWrapper';
...
+    const emailFormElement = root.container.querySelector('[data-e2e="email-form"]');
+    expect(root.container.querySelector('[data-e2e="stream-response-loader"]')).toBeVisible();
+    expect(userFormElement).toBeNull();
+    expect(emailFormElement).toBeNull();
+
+    const { chat } = root.store.getState();
+    expect(chat.isLoading).toBe(true);
+    expect(historyElements.length).toEqual(chat.historyIds.length);
+  });
+
```
Comment (venzovisa) [inline __tests__/feature/chat-history.spec.jsx:69]: @IlievaDayana Optional refactor:
` const { chat: { isLoading, historyIds } } = root.store.getState();
    expect(isLoading).toBe(true);
    expect(historyElements.length).toEqual(historyIds.length);
`

File: __tests__/feature/chat-history.spec.jsx [hunk 4]
```diff
@@ -104,4 +139,208 @@ describe('Chat-history event works and visualizes items accordingly', () => {
...
+    expect(userFormElement).toBeNull();
+    expect(emailFormElement).toBeVisible();
+    expect(loaderElement).toBeNull();
+    expect(ctaElement).toBeNull();
+
+    const { chat } = root.store.getState();
+    expect(chat.isLoading).toBe(false);
+    expect(chat.isStreaming).toBe(false);
+  });
+
```
Comment (venzovisa) [inline __tests__/feature/chat-history.spec.jsx:181]: `const { chat: { isLoading,  isStreaming} } = root.store.getState();
    expect(isLoading).toBe(false);
    expect(isStreaming).toBe(false);`

**Review by venzovisa — APPROVED** [review]

**Pull request merged**